pennylane/qchem/openfermion_obs.py

```diff
@@ -1077,11 +1066,14 @@ def _molecular_hamiltonian(
             raise ValueError(
                 "Only 'jordan_wigner' mapping is supported for the differentiable workflow."
             )
-        if mult != 1:
+        n_electrons = sum([atomic_numbers[s] for s in symbols]) - charge
+
+        if n_electrons % 2 == 1 or mult != 1:
             raise ValueError(
-                "Openshell systems are not supported for the differentiable workflow. Use "
-                "`method = 'pyscf'` or change the charge or spin multiplicity of the molecule."
+                "Open-shell systems are not supported for the requested backend. Use "
+                "method = 'openfermion' or change the charge or spin multiplicity of the molecule."
```

Comment:
We don't support open-shell molecules for `method=pyscf` as well? 🤔

Reply:
We can generate molecular_hamiltonian for open-shell molecules with pyscf backend.

Reply:
This should say both pyscf and openfermion backends